packages/thirdweb/src/extensions/prebuilts/deploy-modular-core.test.ts

```diff
@@ -0,0 +1,156 @@
+import { ANVIL_CHAIN } from "../../../test/src/chains.js";
+import { TEST_CLIENT } from "../../../test/src/test-clients.js";
+import { TEST_ACCOUNT_A } from "../../../test/src/test-wallets.js";
+import { getContract } from "../../contract/contract.js";
+import { sendTransaction } from "../../transaction/actions/send-transaction.js";
+import { installPublishedExtension } from "../modular/ModularCore/write/installPublishedExtension.js";
+import { uninstallExtensionByProxy } from "../modular/ModularCore/write/uninstallExtensionByProxy.js";
+import { uninstallPublishedExtension } from "../modular/ModularCore/write/uninstallPublishedExtension.js";
+import { getInstalledExtensions } from "../modular/__generated__/ModularCore/read/getInstalledExtensions.js";
+import { deployPublishedContract } from "./deploy-published.js";
+
+describe.runIf(process.env.TW_SECRET_KEY)(
+  "deployModularCore",
+  {
+    timeout: 120000,
+  },
+  () => {
+    it("should deploy a published autofactory modular core contract", async () => {
+      const address = await deployPublishedContract({
+        client: TEST_CLIENT,
+        chain: ANVIL_CHAIN,
+        account: TEST_ACCOUNT_A,
+        contractId: "DemoCore",
+        contractParams: [TEST_ACCOUNT_A.address, [], ["0x"]],
+        publisher: "0xFD78F7E2dF2B8c3D5bff0413c96f3237500898B3",
+      });
+      expect(address).toBeDefined();
+      expect(address.length).toBe(42);
+
+      const installedExtensions = await getInstalledExtensions({
+        contract: getContract({
+          client: TEST_CLIENT,
+          chain: ANVIL_CHAIN,
+          address,
+        }),
+      });
+      expect(installedExtensions.length).toBe(0);
+    });
+
+    it("should uninstall an extension by proxy address", async () => {
+      // deploy core contract with extension
+      const address = await deployPublishedContract({
+        client: TEST_CLIENT,
+        chain: ANVIL_CHAIN,
+        account: TEST_ACCOUNT_A,
+        contractId: "DemoCore",
+        contractParams: [TEST_ACCOUNT_A.address, [], ["0x"]],
+        publisher: "0xFD78F7E2dF2B8c3D5bff0413c96f3237500898B3",
+      });
+
+      const core = getContract({
+        client: TEST_CLIENT,
+        chain: ANVIL_CHAIN,
+        address,
+      });
+
+      // install extension with published name
+      const installTransaction = installPublishedExtension({
+        contract: core,
+        chain: ANVIL_CHAIN,
+        client: TEST_CLIENT,
+        account: TEST_ACCOUNT_A,
+        extensionName: "DemoExtensionWithFunctions",
+        publisherAddress: "0xFD78F7E2dF2B8c3D5bff0413c96f3237500898B3",
+      });
+      await sendTransaction({
+        transaction: installTransaction,
+        account: TEST_ACCOUNT_A,
+      });
+
+      // get installed extension proxy address
+      let installedExtensions = await getInstalledExtensions({
+        contract: core,
+      });
+      const extensionAddress = installedExtensions[0]?.implementation;
```

Comment:
add an `expect()` here to verify the extension is installed and returned from `getInstalledExtensions` etc